File: java/djjavawork/HelloJava06/src/studyjava05_abstract/Food.java
package studyjava05_abstract;

public abstract class Food {

	private int price;
	private String name;
	public int getPrice() {
		return price;
	}
	public void setPrice(int price) {
		this.price = price;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	//음식이라 하면, 먹는 방법이 있다.
	//음식이라는 이름의 음식은 없다.
	//다만... 음식을 상속받았다면 반드시 eat은 구현을 해야 한다.
	public abstract void eat();
}

File: djCSharp/HelloCSharp05/HelloCSharp05_03_Inheritance/Food.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HelloCSharp05_03_Inheritance
{
    //추상화된 개념을 만들 때 주로 쓰임
    //Food a = new Food(); //불가능
    //Food a = new Gimbap();//단 Gimbap이
    //Food의 후손이어야 함
    public abstract class Food
    {
        public string name { set; get; }
        public int price { get; set; }
        public abstract void Eat();
        public void leaveFood()
        {
            Console.WriteLine(name+"못먹겠다.");
            Console.WriteLine(price+"값이 아깝다.");
        }

    }
}
